server/embed/config.go

```diff
@@ -455,6 +459,9 @@ type Config struct {
 
 	// V2Deprecation describes phase of API & Storage V2 support
 	V2Deprecation config.V2DeprecationEnum `json:"v2-deprecation"`
+
+	// ServerFeatureGate is a server level feature gate
+	ServerFeatureGate featuregate.FeatureGate
```

Comment:
```suggestion
	ServerFeatureGate featuregate.MutableFeatureGate
```

Reply:
I think it is safer to keep the general interface immutable, and explicitly cast to `MutableFeatureGate` when it is needed in a limited few places. This way, we can prevent the feature gate from being accidentally changed.